python_modules/dagster/dagster/_daemon/backfill.py

```diff
@@ -10,6 +10,20 @@
 from dagster._utils.error import SerializableErrorInfo
 
 
+class BackfillLogAdapter(logging.LoggerAdapter):
+    """Logger that will always prepend the backfill id to the log message.
+
+    Usage:
+
+    :code-block: python
+
+        adapted_backfill_logger = BackfillLogAdapter(logger, {"backfill_id": backfill_id})
+    """
+
+    def process(self, msg, kwargs) -> Any:  # -> Any:
+        return f"Backfill {self.extra['backfill_id']}: {msg}", kwargs
```

Comment:
I like the idea but I'm not sure about this specific approach

when the logging output format is set to json, the ideal outcome is that the backfill id is set as a key value pair

Reply:
dumb question - how do i set the log output to json?

Reply:
`dagster-daemon` has `--log-format` cli option 
i think we may do something clever for how we configure it in the context of running it internally

Reply:
ok I see what you mean and i think i figured out how to do this. we can add the `ExtraAdder` to the processors used by `structlog`. that will take anything passed to the logger like this `logger.info("the message", extra={"foo":"bar"}` and make them key-value pairs in the logs. we can also add the backfill id to the `extra` in the Adapter. it just means for json and rich logging, you get the backfill id in multiple places, the log message and as a key-value pair. I updated the description with the updated log outputs


maybe adding the `extra` in the adapter is still kinda hacky, i'm not familiar at all with logging standards, but this is the only thing i've found that actually works...

Reply:
Maybe a [ContextFilter](https://docs.python.org/3/howto/logging-cookbook.html#using-filters-to-impart-contextual-information) might also work? need to test how the info that gets added interacts with json formatting though. it also looks like you apply it to the main `logger` rather than create a new logger with the adapter. so i'd need to make sure the filter gets removed and reset for each different backfill